lib/version.ex

```diff
@@ -6,20 +6,24 @@ defmodule PaperTrail.Version do
 
   @setter PaperTrail.RepoClient.originator || nil
 
-  @item_type Application.get_env(:paper_trail, :item_type, :integer)
-  @originator_type Application.get_env(:paper_trail, :originator_type, :integer)
+  # @item_type Application.get_env(:paper_trail, :item_type, :integer)
+  # @originator_type Application.get_env(:paper_trail, :originator_type, :integer)
 
   schema "versions" do
+    item_type = Application.get_env(:paper_trail, :item_type, :integer)
+    originator_type = Application.get_env(:paper_trail, :originator_type, :integer)
```

Comment:
All is good, but this part concerns me. Why do we need to put version module attributes to the schema?

Reply:
Pardon me, I don't understand the question. :-)
What I am doing here is removing the use of module attribute for the item_type and originator_type.

Reply:
yes, and what is the reason here? Because I'm not sure if defining variables in the schema is a good convention or necessary.

Reply:
Ah, I see what you mean.

Because when one of the problems I am getting is the `item_type` still returning `:integer` type even though I have set it to be `:uuid` in the config.
Just like the other problems I am experiencing, the runtime value is not being returned with the use of module @ttribute.

Agree with you that this is not a good convention.
I am using variables here because it's the most straightforward way to make the values from `PaperTrail.RepoClient.originator()`, `Application.get_env(:paper_trail, :item_type, :integer)`, and `Application.get_env(:paper_trail, :originator_type, :integer)` reusable by reference.

Otherwise, I can try:

```
  # @setter PaperTrail.RepoClient.originator()
  # @item_type Application.get_env(:paper_trail, :item_type, :integer)
  # @originator_type Application.get_env(:paper_trail, :originator_type, :integer)

  schema "versions" do

    field :event, :string
    field :item_type, :string
    field :item_id, Application.get_env(:paper_trail, :item_type, :integer)
    field :item_changes, :map
    field :originator_id, Application.get_env(:paper_trail, :originator_type, :integer)
    field :origin, :string, read_after_writes: true
    field :meta, :map

    if PaperTrail.RepoClient.originator() do
      belongs_to PaperTrail.RepoClient.originator()[:name], PaperTrail.RepoClient.originator()[:model], define_field: false, foreign_key: :originator_id, type: Application.get_env(:paper_trail, :originator_type, :integer)
    end

    timestamps(updated_at: false)
  end
```

What do you think is a better way to handle this situation?
The use of module attribute as a reference to the values is way better but it is just not returning my config values when I run it.